filters/auth/oauth.go

```diff
@@ -145,9 +149,35 @@ func intersect(left []string, right []string) bool {
 	return false
 }
 
+func createHttpClient(timeout time.Duration) *http.Client {
+	var client *http.Client
+	if timeout != 0 {
+		transport := &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: timeout,
+			}).DialContext,
+		}
+
+		client = &http.Client{Transport: transport}
+
+		go func() {
```

Comment:
We need to have a quit channel to not leak this goroutine

Reply:
At which part of the lifecycle should the `quit` channel be closed?

Reply:
@szuecs  I did not find any lifecycle method in `oauth` filter similar to `Close()` method in the kubernetes ingress controller. I have added the `quit` channel to return from the `goroutine` but still not sure at which point to send to the `quit` channel.

Reply:
We have a sigterm handler

Reply:
I checked the quit channel implementations in those 2 but the `kubernetes/dataclients` and `proxy/proxy.go` have `Close()` methods, but a filter doesn't. Should I implement a custom dataclient with Close() method for the oauth filter?